src/storage/mvcc/reader/point_getter.rs

```diff
@@ -343,6 +352,34 @@ impl<S: Snapshot> PointGetter<S> {
             ))
         }
     }
+
+    /// Load the value from the lock.
+    ///
+    /// The lock belongs to a committed transaction and its commit_ts <= read's start_ts.
+    fn load_data_from_lock(&mut self, user_key: &Key, lock: Lock) -> Result<Option<Value>> {
+        debug_assert!(lock.ts < self.ts && lock.min_commit_ts <= self.ts);
+        match lock.lock_type {
+            LockType::Put => {
+                if self.omit_value {
+                    return Ok(Some(vec![]));
+                }
+                match lock.short_value {
+                    Some(value) => {
+                        // Value is carried in `lock`.
+                        self.statistics.processed_size += user_key.len() + value.len();
+                        Ok(Some(value.to_vec()))
+                    }
+                    None => {
+                        let value = self.load_data_from_default_cf(lock.ts, user_key)?;
+                        self.statistics.processed_size += user_key.len() + value.len();
+                        Ok(Some(value))
+                    }
+                }
+            }
+            LockType::Delete => Ok(None),
+            LockType::Lock | LockType::Pessimistic => unreachable!(),
```

Comment:
If this happens, it can be the fault from the TiKV client. So, I think returning an error should be more reasonable than panicking.

Reply:
It can't happen because only when fails to `Lock::check_ts_conflict()`, this function is called, so the `LockType` can't be lock or  pessimsitic.

Reply:
I see